charts/kong/tests/golden_test.go

```diff
@@ -0,0 +1,97 @@
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/require"
+)
+
+// -update-golden flag can be passed to go test in order to regenerate golden manifests under testdata/golden.
+var update = flag.Bool("update-golden", false, "update golden test output files")
+
+// TestGolden goes through all test *-values.yaml (helm values) files in ci/ directory and `helm template` kong chart with those
+// against various Kubernetes versions. The output for each is expected to be equal to the generated golden file stored
+// under `./testdata/golden/<valuesFileName>_golden.yaml`.
+// When a difference in the output is expected, you can regenerate golden files by passing `-update-golden` flag while
+// invoking this test (or using `make test.golden.update`).
+func TestGolden(t *testing.T) {
+	helm.AddRepo(t, &helm.Options{}, "bitnami", "https://charts.bitnami.com/bitnami")
+	_, err := helm.RunHelmCommandAndGetOutputE(t, &helm.Options{}, "dependencies", "build", "../", "--skip-refresh")
```

Comment:
It would be nice to include the mention of `helm dependencies build` being called and the reason for that.

Reply:
Yes, good point. Added a comment. 👍